src/override/malloc.cc

```diff
@@ -116,6 +116,65 @@ extern "C"
   }
 #endif
 
+#if !defined(SNMALLOC_NO_REALLOCARR)
+  SNMALLOC_EXPORT int
+    SNMALLOC_NAME_MANGLE(reallocarr)(void* ptr, size_t nmemb, size_t size)
+  {
+    int err = errno, r;
+    auto& a = ThreadAlloc::get();
+    bool overflow = false;
+    size_t sz = bits::umul(size, nmemb, overflow);
+    if (sz == 0)
+    {
+      errno = err;
+      return 0;
+    }
+    if (overflow)
+    {
+      errno = err;
+      return EOVERFLOW;
+    }
+
+    if (ptr == nullptr)
+    {
+      ptr = a.alloc(sz);
```

Comment:
I think you have the level of indirection here incorrect. 
```suggestion
    if (*ptr == nullptr)
    {
      *ptr = a.alloc(sz);
```